plugins/catalog-backend-module-gitea/src/providers/GiteaEntityProvider.ts

```diff
@@ -0,0 +1,266 @@
+    }
+
+    const providerConfigs = readGiteaConfigs(configRoot);
+    const integrations = ScmIntegrations.fromConfig(configRoot).gitea;
+    const providers: GiteaEntityProvider[] = [];
+
+    providerConfigs.forEach(providerConfig => {
+      const integration = integrations.byHost(providerConfig.host);
+      if (!integration) {
+        throw new InputError(
+          `No Gitea integration found that matches host ${providerConfig.host}`,
+        );
+      }
+
+      if (!options.schedule && !providerConfig.schedule) {
+        throw new Error(
+          `No schedule provided neither via code nor config for Gitea-provider:${providerConfig.id}.`,
+        );
+      }
+
+      const taskRunner =
+        options.schedule ??
+        options.scheduler!.createScheduledTaskRunner(providerConfig.schedule!);
+
+      providers.push(
+        new GiteaEntityProvider(
+          providerConfig,
+          integration,
+          options.logger,
+          taskRunner,
+        ),
+      );
+    });
+
+    return providers;
+  }
+
+  private constructor(
+    config: GiteaProviderConfig,
+    integration: GiteaIntegration,
+    logger: LoggerService,
+    taskRunner: SchedulerServiceTaskRunner,
+  ) {
+    this.config = config;
+    this.integration = integration;
+    this.logger = logger.child({
+      target: this.getProviderName(),
+    });
+    this.scheduleFn = this.createScheduleFn(taskRunner);
+  }
+
+  getProviderName(): string {
+    return `gitea-provider:${this.config.id}`;
+  }
+
+  async connect(connection: EntityProviderConnection): Promise<void> {
+    this.connection = connection;
+    await this.scheduleFn();
+  }
+
+  private async parseGiteaJsonResponse(
+    response: Response,
+  ): Promise<GiteaProjectQueryResult> {
+    if (!response.ok) {
+      throw new Error(
+        `Failed to fetch Gitea repositories, status: ${response.status}`,
+      );
+    }
+    try {
+      return await response.json();
+    } catch (error) {
+      throw new Error(`Failed to parse Gitea API response: ${error}`);
+    }
+  }
+
+  private createScheduleFn(
+    taskRunner: SchedulerServiceTaskRunner,
+  ): () => Promise<void> {
+    return async () => {
+      const taskId = `${this.getProviderName()}:refresh`;
+      return taskRunner.run({
+        id: taskId,
+        fn: async () => {
+          const logger = this.logger.child({
+            class: GiteaEntityProvider.prototype.constructor.name,
+            taskId,
+            taskInstanceId: uuid.v4(),
+          });
+
+          try {
+            await this.refresh(logger);
+          } catch (error) {
+            logger.error(
+              `${this.getProviderName()} refresh failed, ${error}`,
+              error as Error,
+            );
+          }
+        },
+      });
+    };
+  }
+
+  async refresh(logger: LoggerService): Promise<void> {
+    if (!this.connection) {
+      throw new Error('Gitea discovery connection not initialized');
+    }
+    let allRepos: any[] = [];
```

Comment:
It's unfortunate that this is an `any` - making a type for this would be desirable. Worst case use an array [element type extractor helper](https://stackoverflow.com/a/51399781) out of `GiteaProjectQueryResult`